extensions/amp-bind/0.1/amp-state.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {bindForDoc} from '../../../src/bind';
+import {isJsonScriptTag} from '../../../src/dom';
+import {toggle} from '../../../src/style';
+import {tryParseJson} from '../../../src/json';
+import {user} from '../../../src/log';
+
+export class AmpState extends AMP.BaseElement {
+  /** @override */
+  getPriority() {
+    // Loads after other content.
+    return 1;
+  }
+
+  /** @override */
+  isAlwaysFixed() {
+    return true;
+  }
+
+  /** @override */
+  isLayoutSupported(unusedLayout) {
+    return true;
+  }
+
+  /** @override */
+  buildCallback() {
+    const TAG = this.getName_();
+
+    this.element.setAttribute('aria-hidden', 'true');
+
+    const id = this.element.id;
```

Comment:
nit: const id = user().assert(this.element.id, '%s element must have an id.', TAG);